src/udp.c

```diff
@@ -178,14 +273,20 @@ void udp_receive(int udpsocket, void* ctx)
 
 static __attribute__((constructor)) void udp_init()
```

Comment:
There is a problem with udp_init. Calling add_option_cb does not enforce the set_* methods to be run instantly, quite the opposite, they will only run once parse_cmdline is called from main.c. So dhcpsnoopingd will not listen on the newly configured port but instead udp_init will listen on the default port.